floodlight/io/catapultApi.py

```diff
@@ -0,0 +1,492 @@
+
+    response_ok =200
+    for i in player_ids:
+        url = f'activities/{activity_id}/athletes/{i}/sensor'
+        player_data = requests.get(base_url + url, headers=headers)
+        
+        if player_data.status_code == response_ok:
+            response_data_dict_list.append(player_data.json()[0])
+        else:
+            raise APIRequestError(f"API request failed with status code {player_data.status_code}")
+
+    return response_data_dict_list
+
+def get_key_names_from_dict_list(response_data_dict_list: List[Dict]) -> set:
+    """
+    Extracts unique key names from a list of response_data.
+
+    Parameters
+    ----------
+    response_data_dict_list : List[Dict]
+        List of response_data.
+
+    Returns
+    -------
+    set
+        A set of unique key names.
+    """
+    
+    mapping, _, _ = get_mappings()
+
+    recorded_keys = set()
+
+    for athlete_entry in response_data_dict_list:
+        recorded_keys.update(key for key in athlete_entry.keys() if key in mapping)
+        if len(athlete_entry["data"]) != 0:
+            recorded_keys.update(key for key in athlete_entry["data"][0].keys() if key in mapping)
+
+    return recorded_keys
+
+def get_meta_data(response_data_dict_list: List[Dict], coord_format: str = "xy") -> Tuple[Dict[str, Dict[str, List[str]]], int, int, int]:
+    """
+    Extracts metadata from the response_data.
+
+    Parameters
+    ----------
+    response_data_dict_list : List[Dict]
+        List of response_data.
+
+    Returns
+    -------
+    Tuple[Dict[str, Dict[str, List[str]]], int, int, int]
+        - Dictionary mapping group IDs to player identifiers.
+        - Number of frames.
+        - Framerate.
+        - Null timestamp.
+    """
+
+    mapping, reversed_group_mapping, reversed_sensor_mapping = get_mappings()
+    
+    key_names = list(get_key_names_from_dict_list(response_data_dict_list))
+    key_names_mapped = [mapping[i] for i in key_names]
+    sensor_identifier = {"name", "number", "sensor_id", "mapped_id"}
+    key_names_mapped_set = set(key_names_mapped)
+
+    recorded_sensor_identifier = list(key_names_mapped_set & sensor_identifier)
+    sensor_links = {
+        key: index
+        for (index,key) in enumerate(key_names_mapped) #.items()
+        if key in recorded_sensor_identifier
+    }
+    group_identifier_set = {"group_id", "group_name"}
+    recorded_group_identifier = list(key_names_mapped_set & group_identifier_set)
+    pID_dict = {}
+    t = []
+    has_groups = len(recorded_group_identifier) > 0
+    if not has_groups:
+        warnings.warn("Since no group exist in data, dummy group '0' is created!")
+
+    max_count = 0
+    for athlete_entry in response_data_dict_list:
+        athlete_t = [int(entry["ts"]) for entry in athlete_entry["data"]]
+        t.extend([int(entry["ts"]) * 100 + int(entry["cs"]) for entry in athlete_entry["data"]]) # timestamps are in centiseconds. Magic number 100 is needed for conversion to centiseconds

```

Comment:
I think this should be rephrased. The value in `entry["ts"]` is the timestamp in seconds. The value in `entry["cs"]` is the centisecond division of that timestamp. The magic number is used to merge the two values on the same scale.